docs/developer/best-practices/security.asciidoc

```diff
@@ -1,55 +1,128 @@
+Application Security Project (OWASP) references to learn more about these types of attacks.
+
+=== Cross-site Scripting (XSS) ===
+
+https://owasp.org/www-community/attacks/xss[_OWASP reference for XSS_]
+
+XSS is a class of attacks where malicious scripts are injected into vulnerable websites. {kib} defends against this primarily by widespread
+use of the React framework to safely encode data that is rendered to pages, widespread use of the EUI framework to
+https://elastic.github.io/eui/#/navigation/link#link-validation[automatically sanitize links], and a restrictive Content-Security-Policy
+header.
+
+*Best practices:*
+
+* Check for usage of dangerous functions or assignments that can result in unescaped user input in the browser DOM. Avoid
+using these at all costs:
+** *React:* https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml[`dangerouslySetInnerHtml`].
+** *Browser DOM:* `Element.innerHTML` and `Element.outerHTML`.
+** *jQuery:* `$.html`, `$.append`, `$.appendTo`, `$.prepend`, and `$.prependTo`. Consider using `$.text` instead, which is a safe
+alternative to these functions.
+* If using any of the aforementioned unsafe functions/assignments is absolutely necessary, follow
+https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#xss-prevention-rules[these XSS prevention
+rules] to ensure that user input is not inserted into unsafe locations and to make sure that it is escaped properly.
+* Use EUI components to build your UI whenever possible, particularly when rendering `href` links. Otherwise, sanitize user input before
+rendering links to ensure that they do not use the `javascript:` protocol.
+
+=== Cross-site Request Forgery (CSRF/XSRF) ===
+
+https://owasp.org/www-community/attacks/csrf[_OWASP reference for CSRF_]
+
+CSRF is a class of attacks where a user is forced to execute an action on a vulnerable website that they're logged into, usually without
+their knowledge. {kib} defends against this by requiring
+https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers[custom
+request headers] for API endpoints. For more information, see <<api-request-headers, API Request Headers>>.
+
+*Best practices:*
+
+* Ensure all HTTP routes are registered via the <<http-service, {kib} HTTP service>> to take advantage of the aforementioned custom request
+header security control.
+
+=== Remote Code Execution (RCE) ===
+
+https://owasp.org/www-community/attacks/Command_Injection[_OWASP reference for Command Injection_],
+https://owasp.org/www-community/attacks/Code_Injection[_OWASP reference for Code Injection_]
+
+RCE is a class of attacks where an attacker executes malicious code or commands on a vulnerable server. {kib} defends against this by using
+ESLint rules to restrict vulnerable functions, and by hooking into / hardening usage of these in third-party dependencies.
+
+*Best practices:*
```

Comment:
Should we mention being extra careful when spawning child processes using any parameters that are user controllable?

Reply:
Good idea, I can add that in.